ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -14033,12 +14033,26 @@ ABoundActivity * HqlCppTranslator::doBuildActivityDedup(BuildCtx & ctx, IHqlExpr
 
     if (!useHash)
     {
-        if (info.compareAllRows)
-            doBuildBoolFunction(instance->classctx, "compareAll", info.compareAllRows);
-        if (!info.keepLeft)
-            doBuildBoolFunction(instance->classctx, "keepLeft", info.keepLeft);
+        // Only necessary to create getFlags() if different from base class getFlags() - i.e. different from default
+        if ( !(!info.compareAllRows && !info.keepBest && info.keepLeft) )
```

Comment:
The double negative is a bit confusing, would read better as:
```
if (info.compareAllRows || info.keepBest || !info.keepLeft)
```

But in either case is the !info.keepLeft correct?
Since below you do:
```
           if (info.keepLeft)
                flags.append("|DFkeepleft");
```